kernel/src/syscall.rs

```diff
@@ -389,6 +395,11 @@ pub trait UserspaceKernelBoundary {
     /// initialization must happen in the `initialize_process()` function.
     type StoredState: Default;
 
+    /// Called by the kernel during process creation. Allows for
+    /// architecture-specific process layout decisions, such as stack pointer
+    /// initialization.
+    fn initial_process_layout(&self, process_memory_start: *const u8) -> InitialProcessLayout;
```

Comment:
```suggestion
    /// Called by the kernel during process creation. Allows for
    /// architecture-specific process layout decisions, such as stack pointer
    /// initialization.
    ///
    /// Returns two pointers. First, the address where the kernel should set the start of the
    /// process's stack, and second the address where the kernel should set the app_brk.
    fn initial_process_layout(&self, process_memory_start: *const u8) -> (*const u8, *const u8);
```

Reply:
I used your new comment. Is there a reason you suggested removing the `InitialProcessLayout` struct? Is that to prepare for changes that make this only return the initial break?

Reply:
Oh right I meant to comment on that. I think the return type is fine in general, it just is a little inconsistent with the remainder of the trait, specifically the return value for switch_to_process().

Reply:
Okay, making it more consistent with the rest of the trait seems good to me. I'll go ahead and make that change.

Reply:
There's actually another problem: we can't actually return a pointer from this new init function. We have a bit of a dependency chain: the syscall impl can't setup any process memory until memory has actually been allocated for that process, but Process cannot allocate any memory for a process until it asks the MPU what it can actually protect, and Process cannot call the MPU functions before it knows how much RAM the process needs.

So, Process has to ask syscall for a number of bytes, use that to calculate the requirements for the process, then ask the MPU, then create the actual pointers/addresses to initial process memory.

I'm working on a 1.x implementation to work all of this out.